federation/cluster/common.sh

```diff
@@ -224,58 +223,68 @@ function create-federation-api-objects {
 
 # Required
 # FEDERATION_PUSH_REPO_BASE: the docker repo where federated images will be pushed
-
-# Optional
-# FEDERATION_IMAGE_TAG: push all federated images with this tag. Used for ci testing
+# FEDERATION_IMAGE_TAG: the tag of the image to be pushed
 function push-federation-images {
     : "${FEDERATION_PUSH_REPO_BASE?Must set FEDERATION_PUSH_REPO_BASE env var}"
-    local FEDERATION_BINARIES=${FEDERATION_BINARIES:-"federation-apiserver federation-controller-manager"}
+    : "${FEDERATION_IMAGE_TAG?Must set FEDERATION_IMAGE_TAG env var}"
 
-    local imageFolder="${KUBE_ROOT}/_output/${KUBE_BUILD_STAGE}/server/${KUBE_PLATFORM}-${KUBE_ARCH}/kubernetes/server/bin"
+    source "${KUBE_ROOT}/build/common.sh"
+    source "${KUBE_ROOT}/hack/lib/util.sh"
 
-    if [[ ! -d "$imageFolder" ]];then
-	echo "${imageFolder} does not exist! Run make quick-release or make release"
-	exit 1
+    local FEDERATION_BINARIES=${FEDERATION_BINARIES:-"hyperkube"}
+
+    local bin_dir="${KUBE_ROOT}/_output/${KUBE_BUILD_STAGE}/server/${KUBE_PLATFORM}-${KUBE_ARCH}/kubernetes/server/bin"
+
+    if [[ ! -d "${bin_dir}" ]];then
+        echo "${bin_dir} does not exist! Run make quick-release or make release"
+        exit 1
     fi
 
-    for binary in $FEDERATION_BINARIES;do
-	local imageFile="${imageFolder}/${binary}.tar"
+    for binary in ${FEDERATION_BINARIES}; do
+        local bin_path="${bin_dir}/${binary}"
 
-	if [[ ! -f "$imageFile" ]];then
-	    echo "${imageFile} does not exist!"
-	    exit 1
-	fi
+        if [[ ! -f "${bin_path}" ]]; then
+            echo "${bin_path} does not exist!"
+            exit 1
+        fi
 
-	echo "Load: ${imageFile}"
-	# Load the image. Trust we know what it's called, as docker load provides no help there :(
-	docker load < "${imageFile}"
+        local docker_build_path="${bin_path}.dockerbuild"
+        local docker_file_path="${docker_build_path}/Dockerfile"
 
-	local srcImageTag="$(cat ${imageFolder}/${binary}.docker_tag)"
-	local dstImageTag="${FEDERATION_IMAGE_TAG:-$srcImageTag}"
-	local srcImageName="${FEDERATION_IMAGE_REPO_BASE}/${binary}:${srcImageTag}"
-	local dstImageName="${FEDERATION_PUSH_REPO_BASE}/${binary}:${dstImageTag}"
+        rm -rf ${docker_build_path}
+        mkdir -p ${docker_build_path}
 
-	echo "Tag: ${srcImageName} --> ${dstImageName}"
-	docker tag -f "$srcImageName" "$dstImageName"
+        ln "${bin_path}" "${docker_build_path}/${binary}"
+        printf " FROM busybox \n ADD ${binary} /usr/local/bin/${binary}\n" > ${docker_file_path}
```

Comment:
`FROM busybox:glibc` since hyperkube is dynamically linked.
Also, this should be built for all architectures: (IIRC)
 - armel/busybox:glibc
 - aarch64/busybox:glibc
 - ppc64le/busybox:glibc